servatrice/src/servatrice.cpp

```diff
@@ -462,7 +462,12 @@ void Servatrice::scheduleShutdown(const QString &reason, int minutes)
     if (minutes > 0) {
         shutdownTimer = new QTimer;
         connect(shutdownTimer, SIGNAL(timeout()), this, SLOT(shutdownTimeout()));
-        shutdownTimer->start(60000);
+        if (minutes <= 5 || isFirstShutdownMessage) {
+            shutdownTimer->start(60000);
+            isFirstShutdownMessage = false;
+        }
+        else
+            shutdownTimer->start((shutdownMinutes / 2) * 60 * 1000);
```

Comment:
Each time we call this method we create `shutdownTimer = new QTimer;` Which will remove the previous time config. If its the first call we set to 60s, then after that, we re-create the object and assign a new time. Hope that makes sence? It still needs testing, though It shouldnt popup once a min, unless if the shutdown is <= 5m.